src/firmware/guest/mod.rs

```diff
@@ -85,22 +86,21 @@ impl Firmware {
         &mut self,
         message_version: Option<u8>,
         ext_report_request: SnpExtReportReq,
```

Comment:
with this, the end-user still has to fill `certs_address` and `certs_len`, which is a bit C-like which we'll prefer to hide.

Reply:
we might want to get only the data field from the user; we can allocate a certs buffer inside the function.

Reply:
Right, really only looking for a list of certs.

Reply:
Yeah, I can definitely push a commit to make this change.